### ISSUE TITLE
Mail contact didn't run successfully in 1846 if bought in on select routes step

### ISSUE THREAD

**Issue by wmspringer** [tobymao/18xx#3243]
https://18xx.games/game/21637?action=184

OR 1.2
B&O operates. I purchased Mail Contact right before running my trains, but it didn't show up (I didn't get the revenue for it and it didn't show in the history). I had to pay out, then use the Undo button to go back to train selection, then it showed up.

**Comment by crericha** [comment]
I cloned the game and was unsuccessful in reproducing the issue.

I backed up to just before the mail contract purchase and then played through to the dividend step. The mail contract was included in both the log and revenue.

I also stepped through the history of the original game and see the mail contract added to one of the 2 train routes.

I think this is fixed by another commit. Any objection to me closing?
